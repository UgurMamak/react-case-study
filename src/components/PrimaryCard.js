import {Button, Card} from "react-bootstrap";
import {BsArrowDownShort, BsArrowUpShort, BsXCircle} from "react-icons/bs";
import React from 'react';

const PrimaryCard = ({item, deleteCardConfirm, increasePoints, receivedData, decreasePoints}) => {
    return (
        <Card className="primary-card">
            <Card.Header>
                <div className="card-info">
                    <span className="number">{item.points}</span>
                    <span className="text">POINTS</span>
                    <span className="updated-item">Updated Date: {item.updated}</span>
                </div>
            </Card.Header>
            <Card.Body>
                <Card.Title>
                    {item.name}
                    <Button onClick={() => {
                        deleteCardConfirm(item);
                    }}
                            className="delete-btn"><BsXCircle/></Button>
                </Card.Title>
                <div className="short-info">(https://news.ycombinator.com)</div>
                <Card.Footer>
                    <Button onClick={() => {
                        increasePoints(item);
                        receivedData();
                    }}
                            variant="primary"><BsArrowUpShort/> Up vote</Button>
                    <Button onClick={() => {
                        decreasePoints(item);
                        receivedData();
                    }}
                            variant="primary"><BsArrowDownShort/> Down Vote</Button>
                </Card.Footer>
            </Card.Body>
        </Card>
    );
};

export default PrimaryCard;