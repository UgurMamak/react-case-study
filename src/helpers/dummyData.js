export function getDummyData(){
   return  JSON.parse(localStorage.getItem('data'));
}

export function setDummyData(data){
   return localStorage.setItem('data',JSON.stringify(data));
}