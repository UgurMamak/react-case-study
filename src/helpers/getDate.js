export function getDateNow(currentdate) {
   const convertDate= new Date(currentdate)
    return convertDate.getMonth() + 1 + "." +
        convertDate.getDate() + "." +
        convertDate.getFullYear() + " " +
        convertDate.getHours() + ":" +
        convertDate.getMinutes() + ":" +
        convertDate.getSeconds();

    /*return currentdate.getMonth() + 1 + "." +
        currentdate.getDate() + "." +
        currentdate.getFullYear() + " " +
        currentdate.getHours() + ":" +
        currentdate.getMinutes() + ":" +
        currentdate.getSeconds();*/
}
