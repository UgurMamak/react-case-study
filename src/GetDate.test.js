import {getDateNow} from "./helpers/getDate";
import renderer from 'react-test-renderer';
import PrimaryCard from "./components/PrimaryCard";
import React from "react";

test("getdate test",()=>{
    const data = '7.2.2021 17:50:23';
    const result = '7.2.2021 17:50:23';
    console.log(data);
    expect(getDateNow(data)).toEqual(result);
});

describe('App', () => {
    test('snapshot renders', () => {
        const item={
            updated:getDateNow(new Date()),
            created:getDateNow(new Date()),
            points:6,
            key:1,
        }
        const component = renderer.create(<PrimaryCard
            increasePoints={''}
            decreasePoints={''}
            receivedData={''}
            deleteCardConfirm={''}
            item={item} />).toJSON();
        expect(component).toMatchSnapshot();
    });
});