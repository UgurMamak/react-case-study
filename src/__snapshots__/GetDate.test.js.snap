// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div
  className="primary-card card"
>
  <div
    className="card-header"
  >
    <div
      className="card-info"
    >
      <span>
        Updated Date: 
        7.6.2021 17:41:55
      </span>
      <span
        className="number"
      >
        6
      </span>
      <span
        className="text"
      >
        POINTS
      </span>
    </div>
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <button
        className="delete-btn btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M11.854 4.146a.5.5 0 010 .708l-7 7a.5.5 0 01-.708-.708l7-7a.5.5 0 01.708 0z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M4.146 4.146a.5.5 0 000 .708l7 7a.5.5 0 00.708-.708l-7-7a.5.5 0 00-.708 0z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      className="short-info"
    >
      (https://news.ycombinator.com)
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 5.5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V6a.5.5 0 01.5-.5z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M7.646 4.646a.5.5 0 01.708 0l3 3a.5.5 0 01-.708.708L8 5.707 5.354 8.354a.5.5 0 11-.708-.708l3-3z"
            fillRule="evenodd"
          />
        </svg>
         Up vote
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 7.646a.5.5 0 01.708 0L8 10.293l2.646-2.647a.5.5 0 01.708.708l-3 3a.5.5 0 01-.708 0l-3-3a.5.5 0 010-.708z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M8 4.5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V5a.5.5 0 01.5-.5z"
            fillRule="evenodd"
          />
        </svg>
         Down Vote
      </button>
    </div>
  </div>
</div>
`;

exports[`card test 1`] = `
<div
  className="primary-card card"
>
  <div
    className="card-header"
  >
    <div
      className="card-info"
    >
      <span>
        Updated Date: 
      </span>
      <span
        className="number"
      />
      <span
        className="text"
      >
        POINTS
      </span>
    </div>
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      <button
        className="delete-btn btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 15A7 7 0 108 1a7 7 0 000 14zm0 1A8 8 0 108 0a8 8 0 000 16z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M11.854 4.146a.5.5 0 010 .708l-7 7a.5.5 0 01-.708-.708l7-7a.5.5 0 01.708 0z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M4.146 4.146a.5.5 0 000 .708l7 7a.5.5 0 00.708-.708l-7-7a.5.5 0 00-.708 0z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
    <div
      className="short-info"
    >
      (https://news.ycombinator.com)
    </div>
    <div
      className="card-footer"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 5.5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V6a.5.5 0 01.5-.5z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M7.646 4.646a.5.5 0 01.708 0l3 3a.5.5 0 01-.708.708L8 5.707 5.354 8.354a.5.5 0 11-.708-.708l3-3z"
            fillRule="evenodd"
          />
        </svg>
         Up vote
      </button>
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.646 7.646a.5.5 0 01.708 0L8 10.293l2.646-2.647a.5.5 0 01.708.708l-3 3a.5.5 0 01-.708 0l-3-3a.5.5 0 010-.708z"
            fillRule="evenodd"
          />
          <path
            clipRule="evenodd"
            d="M8 4.5a.5.5 0 01.5.5v5a.5.5 0 01-1 0V5a.5.5 0 01.5-.5z"
            fillRule="evenodd"
          />
        </svg>
         Down Vote
      </button>
    </div>
  </div>
</div>
`;
